libc-bottom-half/sources/preopens.c

```diff
@@ -32,6 +29,10 @@ static preopen *preopens;
 static size_t num_preopens;
 static size_t preopen_capacity;
 
+/// Access to the the above preopen must be protected in the presence of
+/// threads.
+static volatile int lock[1];
```

Comment:
Put this inside of `ifdef _REENTRANT` .. also perhaps `#define` LOCK and UNLOCK to no-op in the else case?

i.e. do the same thing we did for `__wasilibc_cwd_unlock`?

Reply:
`LOCK` and `UNLOCK` are already defined this way; I guess I'll add `#ifdef _REENTRANT` around this `lock` but it seems a bit weird since in the no-threads case `lock` will not be defined at all.

Reply:
Oh great, i forget LOCK/UNLOCK were already defined like that.

Reply:
I that case I guess `__wasilibc_cwd_lock/unlock` are not actually needed at all .. no harm I guess in naming the macro but we could just use LOCK/UNLOCK